### ISSUE TITLE
duplicate records in join collection

### ISSUE THREAD

**Issue by anasyb** [balderdashy/sails-mongo#288]
somehow magically I am getting this into the association join collection: 
```
/* 76 */
{
    "_id" : ObjectId("55692607090ba90100d3cc8d"),
    "r_r_r" : ObjectId("000000000000000000000043"),
    "t_r" : ObjectId("000000000000000000000012")
}

/* 77 */
{
    "_id" : ObjectId("55692607090ba90100d3cc8e"),
    "r_r_r" : ObjectId("000000000000000000000043"),
    "t_r" : ObjectId("000000000000000000000012")
}

/* 78 */
{
    "_id" : ObjectId("55692607090ba90100d3cc8f"),
    "r_r_r" : ObjectId("000000000000000000000043"),
    "t_r" : ObjectId("000000000000000000000012")
}

/* 79 */
{
    "_id" : ObjectId("55692607090ba90100d3cc91"),
    "r_r_r" : ObjectId("000000000000000000000042"),
    "t_r" : ObjectId("000000000000000000000011")
}

/* 80 */
{
    "_id" : ObjectId("55692607090ba90100d3cc92"),
    "r_r_r" : ObjectId("000000000000000000000042"),
    "t_r" : ObjectId("000000000000000000000011")
}
```

Although I am seeing this while trying to figure what's going on with some error message (of different IDs pairs)... what this tells me is that waterline isn't telling mongo that these records should be unique... otherwise mongo would have refused to persist them.

**Comment by sailsbot** [comment]
Thanks for posting, @anasyb.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
